File: 240506/행복한 수열의 개수/number-of-happy-sequence.py
n, m = map(int, input().split())
arr = []
for _ in range(n):
    arr.append(list(map(int, input().split())))
ans = 0

def find(arr, m):
    global ans
    for a in arr:
        x = a[0]
        max_cnt = 1
        cnt = 1
        for i in range(1, len(a)):
            if a[i] == x:
                cnt += 1
            else:
                x = a[i]
                cnt = 1
            max_cnt = max(max_cnt, cnt)

        if max_cnt >= m:
            ans += 1

find(arr, m)

col_arr = []
for i in range(n):
    col_in = []
    for j in range(n):
        col_in.append(arr[j][i])
    col_arr.append(col_in)

find(col_arr, m)

print(ans)

File: 260317/행복한 수열의 개수/number-of-happy-sequence.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        
        Scanner sc = new Scanner(System.in);
        
        int n = sc.nextInt();
        int m = sc.nextInt();

        int[][] arr = new int[n][n];

        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j <n; j++)
            {
                arr[i][j] = sc.nextInt();
            }
        }
        int ans = 0;

        // 가로
        for (int i = 0; i < n; i++)
        {
            int a = 0;
            int b = 0;
            int cnt = 1;
            int maxCnt = 0;
            while (b < n)
            {
                b++;
                if (b == n)
                {
                    maxCnt = Math.max(maxCnt, cnt);
                    break;
                }
                
                if (arr[i][a] == arr[i][b])
                {
                    cnt++;

                }
                else 
                {
                    a = b;
                    maxCnt = Math.max(maxCnt, cnt);
                    cnt = 1;
                }
            }
            if (maxCnt >= m)
            {
                ans++;
            }
        }

        // 세로
        for (int i = 0; i < n; i++)
        {
            int a = 0;
            int b = 0;
            int cnt = 1;
            int maxCnt = 0;
            while (b < n)
            {
                b++;
                if (b == n)
                {
                    maxCnt = Math.max(maxCnt, cnt);
                    break;
                }

                if (arr[a][i] == arr[b][i])
                {
                    cnt++;

                }
                else 
                {
                    a = b;
                    maxCnt = Math.max(maxCnt, cnt);
                    cnt = 1;
                }
            }
            if (maxCnt >= m)
            {
                ans++;
            }
        }
        
        System.out.print(ans);

    }
}